### PR TITLE
Historical finpeople data

### PR DESCRIPTION
**Pull request by kevinverhoff**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."
update the finance/accounting view to include pm data. It also corrects a duplicate additional earnings issue

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [x] <kbd>Format</kbd> has been run on all modified files

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206706431765782

### CODE REVIEW

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql [hunk 1]
```diff
@@ -1,518 +1,440 @@
...
+            od.measurement_name,
+            od.text_box,
+            od.rn_submission,
+
+            os.etr_score as etr_score,
+            os.so_score as so_score,
+
+            if(
+                od.academic_year <= 2024, os.overall_score, od.overall_score
+            ) as overall_score,
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql:325]: @kevinverhoff @cristinabaldor is this year correct?

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql:1]: split up into overall scores and scores + measurements?

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                when pm_term = 'PM4'
+                then date(academic_year + 1, 5, 15)
+            end as eval_date,
+        from
+            {{
+                source(
+                    "performance_management",
+                    "src_performance_management__scores_overall_archive",
+                )
+            }}
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:29]: replace with new staging table

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            overall_tier,
+            case
+                when form_term = 'PM1'
+                then date(academic_year, 10, 1)
+                when form_term = 'PM2'
+                then date(academic_year + 1, 1, 1)
+                when form_term = 'PM3'
+                then date(academic_year + 1, 3, 1)
+            end as eval_date,
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:51]: circular dependency, we need to refactor the upstream views to make this work in one direction

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                then 4
+                when avg(overall_score) >= 2.745
+                then 3
+                when avg(overall_score) >= 1.745
+                then 2
+                when avg(overall_score) < 1.75
+                then 1
+            end as overall_tier,
+            date(academic_year + 1, 5, 15) as eval_date,
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:94]: circular dependency, we need to refactor the upstream views to make this work in one direction

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            - (
+                {{
+                    teamster_utils.date_to_fiscal_year(
+                        date_field="worker_termination_date",
+                        start_month=7,
+                        year_source="start",
+                    )
+                }} - s.academic_year
+            )
+    end as years_teaching,
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:152]: this should be calculated on `int_people__years_experience` and brought in at the reporting layer

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql [hunk 6]
```diff
@@ -14,10 +14,20 @@ with
             and extract(year from date_day) >= 2016
     ),
 
+    additional_earnings_clean as (
+        select distinct
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:18]: avoid using distinct. what's getting deduplicated here? can it be done on the staging table?

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql (hunk 6 above)
Comment (cbini) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__finance_accounting_people_model.sql:18]: leave as-is for now, @kevinverhoff needs to follow up w/ HR if there's a data quality issue

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql (hunk 4 above)
Comment (cbini) [reply src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:94]: wrong ref

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql (hunk 3 above)
Comment (cbini) [reply src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:51]: wrong ref

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql
Comment (cbini) [reply src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__observation_details.sql:1]: later

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql (hunk 5 above)
Comment (kevinverhoff) [reply src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql:137]: That case statement is attempting to determine how many years exp/teaching the person had for the given overall score year. ie: in 2018, when they scored a 3, they had 10 years experience; in 2019 it was 11, etc. Unless we change up the years experience model to be long by year, I think the calculation belongs on int_performance_managmenet__overall_scores

**Review by cbini — APPROVED** [review]

**Pull request merged**